src/cloud/lib/stores/nav/store.tsx

```diff
@@ -422,6 +423,40 @@ function useNavStore(pageProps: any): NavContext {
     ]
   )
 
+  const pendingLoads = useRef<Map<string, Promise<SerializedDocWithBookmark>>>(
+    new Map()
+  )
+  const loadedDocs = useRef<Set<string>>(new Set())
+  const loadDoc = useCallback(
+    async (id: string, reload = false) => {
+      const current = docsMap.get(id)
```

Comment:
might be slightly better to rename it slightly to something like `loadedDoc` or something. 
having it next to several `[..].current` is a bit weird to read imo